skyportal/handlers/api/internal/robotic_instruments.py

```diff
@@ -25,9 +27,18 @@ def get(self):
                             Instrument.api_classname_obsplan.isnot(None)
                         )
                     ).all()
+
+                    # we retrieve the list of unique galaxy catalog names here
+                    # and pass them to the frontend_render_info method
+                    # to avoid having to run that query for each instrument
+                    galaxy_catalog_names = session.scalars(
+                        sa.select(Galaxy.catalog_name).distinct()
+                    ).all()
                     retval = {
                         i.id: i.api_class_obsplan.frontend_render_info(
-                            i, self.current_user
+                            i,
+                            self.current_user,
```

Comment:
Any reason this should also be user_or_token?

Reply:
Ok it should you are right, I didn't change anything here. We're just lagging behind but nice catch. Let me fix it here and in a couple more places

Reply:
Done @mcoughlin